pkg/docker/docker.go

```diff
@@ -581,20 +518,56 @@ func (d *stiDocker) CheckImage(name string) (*api.Image, error) {
 	return nil, nil
 }
 
+func base64EncodeAuth(auth dockertypes.AuthConfig) (string, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(auth); err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(buf.Bytes()), nil
+}
+
 // PullImage pulls an image into the local registry
 func (d *stiDocker) PullImage(name string) (*api.Image, error) {
 	name = getImageName(name)
-	err := d.kubeDockerClient.PullImage(name, d.pullAuth, dockertypes.ImagePullOptions{})
+
+	// RegistryAuth is the base64 encoded credentials for the registry
+	base64Auth, err := base64EncodeAuth(d.pullAuth)
 	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	resp, err := d.InspectImage(name)
+	opts := dockertypes.ImagePullOptions{RegistryAuth: base64Auth}
+	ctx := context.Background() // no timeout
+	resp, err := d.client.ImagePull(ctx, name, opts)
 	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	if resp != nil {
+	defer resp.Close()
+	reporter := newProgressReporter(name, func() {})
+	reporter.start()
+	defer reporter.stop()
+	decoder := json.NewDecoder(resp)
+	for {
+		var msg dockermessage.JSONMessage
+		err = decoder.Decode(&msg)
+		if err == io.EOF {
```

Comment:
we've had issues before io.EOF errors get transformed into other types of `EOF` and checks like this don't fire; hence, in this case, we'd end up in a loop that does not exit.

Create a timer (2 minutes per what we do with inspect image) and leverage select on channels so that we either a) timeout after our typical docker interaction timeout or b) break on the EOF

There are examples in other parts of the code of this pattern. (UPDATE: never mind, I see you use the patter in your new progress indicator code  )

Possibly setting the ctx with 2 minutes vs. no timeout might be sufficient as well, but the infinite for loop with no timeout based short circuit still scares me.  

I at least would like to see the protection in both places, but if you disagree certainly bring in others to chime in.